pkg/cloud/aws/actuators/machine/actuator_test.go

```diff
@@ -580,379 +585,115 @@ func TestImmutableStateChange(t *testing.T) {
-
-func getWorkerMachineScope(t *testing.T, mockEC2 ec2iface.EC2API) *actuators.MachineScope {
-	return getTestMachineScope(getWorkerTestMachine(), t, mockEC2)
 }
 
-func getControlplaneMachineScope(t *testing.T, mockEC2 ec2iface.EC2API) *actuators.MachineScope {
-	return getTestMachineScope(getControlplaneTestMachine(), t, mockEC2)
+type configMapsGetter struct {
+	configMap   *v1.ConfigMap
+	getError    error
+	createError error
 }
 
-func getRandoMachineScope(t *testing.T, mockEC2 ec2iface.EC2API) *actuators.MachineScope {
-	return getTestMachineScope(getRandomTestMachine(), t, mockEC2)
+func (c *configMapsGetter) ConfigMaps(namespace string) corev1client.ConfigMapInterface {
+	return &configMapClient{
+		configMap:   c.configMap,
+		getError:    c.getError,
+		createError: c.createError,
+	}
 }
 
-func getInvalidMachineScope(t *testing.T) *actuators.MachineScope {
-	is := getTestMachineScope(getControlplaneTestMachine(), t, nil)
-	is.Cluster = nil
-	return is
+type configMapClient struct {
+	configMap   *v1.ConfigMap
+	getError    error
+	createError error
 }
 
-func getNoLabelMachineScope(t *testing.T, mockEC2 ec2iface.EC2API) *actuators.MachineScope {
-	return getTestMachineScope(getNoLabelTestMachine(), t, mockEC2)
+func (c *configMapClient) Create(configMap *v1.ConfigMap) (*v1.ConfigMap, error) {
+	return c.configMap, c.createError
 }
 
-func getTestControlplaneMachines() []*clusterv1.Machine {
-	return []*clusterv1.Machine{
-		{
-			ObjectMeta: metav1.ObjectMeta{
-				Name:      "master-0",
-				Namespace: "awesome-ns",
-				Labels: map[string]string{
-					"set": "controlplane",
-				},
-			},
-			Spec: clusterv1.MachineSpec{
-				Versions: clusterv1.MachineVersionInfo{
-					Kubelet:      "v1.13.0",
-					ControlPlane: "v1.13.0",
-				},
-			},
-		},
-		{
-			ObjectMeta: metav1.ObjectMeta{
-				Name:      "master-1",
-				Namespace: "awesome-ns",
-				Labels: map[string]string{
-					"set": "controlplane",
-				},
-			},
-			Spec: clusterv1.MachineSpec{
-				Versions: clusterv1.MachineVersionInfo{
-					Kubelet:      "v1.13.0",
-					ControlPlane: "v1.13.0",
-				},
-			},
-		},
+func (c *configMapClient) Get(name string, getOptions metav1.GetOptions) (*v1.ConfigMap, error) {
+	if c.getError != nil {
+		return nil, c.getError
 	}
+	return c.configMap, nil
 }
 
-func getMockEC2APIDescribeInstancesNotFound(ne []string, mockCtrl *gomock.Controller) ec2iface.EC2API {
-	mockEC2 := mock_ec2iface.NewMockEC2API(mockCtrl)
-	for _, n := range ne {
-		dmi := &ec2.DescribeInstancesInput{
-			Filters: []*ec2.Filter{
-				filter.EC2.VPC(""),
-				filter.EC2.ClusterOwned(""),
-				filter.EC2.Name(n),
-				filter.EC2.InstanceStates(ec2.InstanceStateNamePending, ec2.InstanceStateNameRunning),
-			},
-		}
-		mockEC2.EXPECT().DescribeInstances(dmi).Return(
-			nil,
-			awserrors.NewNotFound(fmt.Errorf("mocked not found"))).Times(1)
-	}
-
-	return mockEC2
+func (c *configMapClient) Update(*v1.ConfigMap) (*v1.ConfigMap, error) {
+	panic("not implemented")
```

Comment:
These panics are generally why I scope down interfaces as much as possible. Unfortunately client-go doesn't make it easy and since these objects will be namespaced at run time we'd have to scope both ConfigMapNamespacer and then the ConfigMapGetter which is kind of ugly.

What you've done is great though, this comment needs no action